### PR TITLE
feat(ingress) support terminationGracePeriodSeconds

### PR DESCRIPTION
**Pull request by chazdnato**
#### What this PR does / why we need it:

There are cases where due to network load balancer (NLB) configuration
that a delay needs to be introduced to the containers during shutdown.
This delay is configurable for the Kong Proxy container through the helm
chart, however it is not possible to configure this delay for the
ingress controller. This can result in stale upstreams during NLB
draining and downtime for services. Making this delay configurable seems
the easiest approach to addressing this limitation.

We have a PR for the kong-ingress-controller which will [update its
behaviour](https://github.com/Kong/kubernetes-ingress-controller/pull/2494) so we can configure the termination delay.

This PR adds the ability to set a concomitant value in the helm chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [x] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

**Comment by chazdnato** [comment]
@kong/team-k8s Could we get an update on this? The [related PR](https://github.com/Kong/kubernetes-ingress-controller/pull/2494) has been merged, and we will need this change to the helm chart in order to utilise this feature. Thank you!

File: charts/kong/CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ## Unreleased
 
-Nothing yet.
+* Added terminationDelaySeconds for Ingress Controller.
+  ([597]https://github.com/Kong/charts/pull/597)
```
Comment (shaneutt) [inline charts/kong/CHANGELOG.md]:
```suggestion
  ([597](https://github.com/Kong/charts/pull/597)
```

**Review by shaneutt — COMMENTED** [review]
Looks good, and I think just about ready to go, just a broken link in the changelog.

File: charts/kong/CHANGELOG.md (hunk 1 above)
Comment (chazdnato) [reply charts/kong/CHANGELOG.md]: good catch, thank you! Might need an extra `)` at the end then?

Comment (shaneutt) [reply charts/kong/CHANGELOG.md]: Ooops, yep.

Comment (chazdnato) [reply charts/kong/CHANGELOG.md]: Added in another commit (I presume there will be a squash later?)

Comment (shaneutt) [reply charts/kong/CHANGELOG.md]: No problem, I'll squash.

**Review by shaneutt — APPROVED** [review]

**Pull request merged**